### DISCUSSION TITLE
Redefining type annotation for a member of a class defined in 3rd party library

### DISCUSSION THREAD

**Discussion by dmi-dmi-dmi** [python/typing#1161] [Q&A] [answered]
Hi guys.

I have quite large legacy codebase which uses a 3rd-party framework with type annotations.

Most part of the codebase looks like this

```
from framework import  BaseClass

class CustomHandler(BaseClass):
    def process(self, data):
        params = self.app.registry.get_params()
        self.app.added_in_runtime_property.handle_complex_case(parms, data)    
```

The problem is that `app` has type annotation in the framework code and `added_in_runtime_property` is a property added in runtime through hooks exposed by the framework.

Is there a way to redefine type annotation for `app` so that it also includes `added_in_runtime_property` without actually subclassing `BaseClass` and inheriting from it instead in the codebase?

**Comment by Gobot1234** [comment, marked as answer]
```py
from framework import BaseClass

class CustomHandler(BaseClass):
    app: MyCustomApp

    def process(self, data):
        params = self.app.registry.get_params()
        self.app.added_in_runtime_property.handle_complex_case(parms, data)
```
Should work, app wouldn't do anything to BaseClass.app (which would still be where python looks for the definition)

**Reply by dmi-dmi-dmi** [reply]
@Gobot1234 huh, I didn't think you can do it this way. Thanks a lot!

I am just curious if there is a way to do this for entire codebase without modifying handler files. I mean it would be nice to do so but I am more than fine with your approach as it should help improve quality of the code.
